Sources/Schema.swift

```diff
@@ -100,7 +101,27 @@ extension SchemaBuilder {
         }
 
         let name = components[2]
+
+        if let cachedSchema = ReferenceResolver.cachedSchemaReferences[name] {
+            return Structure(name: name, structure: cachedSchema)
+        }
+        if let resolving = ReferenceResolver.resolvingReferences[name], resolving {
+            return Structure(name: name, structure: .unresolvedReference(pointer.path))
+        }
+        ReferenceResolver.resolvingReferences[name] = true
         let schema = try builder.build(swagger)
+        ReferenceResolver.cachedSchemaReferences[name] = schema
+        ReferenceResolver.resolvingReferences[name] = nil
         return Structure(name: name, structure: schema)
     }
 }
+
+enum ReferenceResolver {
```

Comment:
I really like your generic resolver, though it has some errors in it and would actually throw an error when it encounters cross referencing, which is what this PR is trying to fix.

Maybe you could have a go in your own code to try to get the cross reference test passing and not crashing at runtime. It's trickier than it seems, especially as everything is a struct. Would love to get your take on it